### PR TITLE
[MySQL] TableAdapters to provide FieldConverters and PartitionKeys

### PR DESCRIPTION
**Pull request by Tang8330**
This is the next PR we will need after https://github.com/artie-labs/reader/pull/569 is merged.

This PR adds 2 functions to `TableAdapter` which is:
1. Retrieving PartitionKeys
2. Retrieving FieldConverters

### CODE REVIEW

File: lib/mysql/converters/converters.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package converters
+
+import (
+	"fmt"
+	"github.com/artie-labs/reader/lib/debezium/converters"
+	"github.com/artie-labs/reader/lib/mysql/schema"
+)
+
+func ValueConverterForType(d schema.DataType, opts *schema.Opts) (converters.ValueConverter, error) {
```
Comment (Tang8330) [inline lib/mysql/converters/converters.go:9]: No changes, just moved it.

File: lib/mysql/converters/converters_test.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+import (
+	"github.com/artie-labs/reader/lib/mysql/schema"
+	"github.com/artie-labs/transfer/lib/debezium"
+	"github.com/artie-labs/transfer/lib/typing"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestValueConverterForType(t *testing.T) {
```
Comment (Tang8330) [inline lib/mysql/converters/converters_test.go:11]: No changes, just moved.

File: sources/mysql/adapter/adapter.go [hunk 3]
```diff
@@ -87,61 +87,3 @@ func (m MySQLAdapter) NewIterator() (transformer.RowsIterator, error) {
 func (m MySQLAdapter) PartitionKeys() []string {
 	return m.table.PrimaryKeys
 }
-
-func valueConverterForType(d schema.DataType, opts *schema.Opts) (converters.ValueConverter, error) {
```
Comment (Tang8330) [inline sources/mysql/adapter/adapter.go:91]: Moved.

File: sources/mysql/adapter/adapter_test.go [hunk 4]
```diff
@@ -48,250 +46,3 @@ func TestMySQLAdapter_TopicSuffix(t *testing.T) {
 		assert.Equal(t, tc.expected, adapter.TopicSuffix())
 	}
 }
-
-func TestValueConverterForType(t *testing.T) {
```
Comment (Tang8330) [inline sources/mysql/adapter/adapter_test.go:52]: Moved.

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**